Add documentation (#1)
TakeScoop/terraform-provider-localtime: docs/data-sources/timestamp.md

# Data Source `timestamp`

The timestamp data source allows you to converts a local timestamp (e.g. without timezone info) into one with the specified timezone and layout.

## Example Usages

> ### Use local timezone

Reviewer: I'd trim down this examples to a single canonical one, _maybe_ one simple and one advanced. If a user wants to understand how to use uncommon optional attributes they can look at the attribute reference.

```terraform
# 2020/11/22 11:11:11 -0800
data "localtime_timestamp" "local_timezone" {
  local_time = "2020/11/22 11:11:11"
}
```

### Specify a timezone location

```terraform
# 2020/11/22 11:11:11 -0500
data "localtime_timestamp" "specific_timezone" {
  local_time = "2020/11/22 11:11:11"
  location   = "America/New_York"
}
```

### Customize only the timezone part of layout

```terraform
# 2020/11/22 11:11:11 PST
data "localtime_timestamp" "custom_timezone_layout" {
  local_time      = "2020/11/22 11:11:11"
  timezone_layout = " MST"
}
```

### Customize timestamp layout

```terraform
# 2020-11-22T11:11:11Z08:00
data "localtime_timestamp" "custom_layout" {
  local_time = "2020/11/22 11:11:11"
  layout     = "2006-01-02T15:04:05Z07:00"
}
```

## Argument Reference

> - `local_time` - (Required) The local timestamp without timezone info. It can be in any valid layout.

Reviewer: "Any valid layout" is ambiguous/circular. Give some examples.

> - `location` - (Optional) The location of the timezone where the exported timestamp will be. It must be a valid [tz database name](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones#List). Default to `Local`.

Reviewer: "Local" is not part of the TZ database, it's a [`LoadLocation`](https://golang.org/pkg/time/#example_FixedZone) behavior. Probably best not to expose that and supply "Local" internally if this field is not set.

> - `layout` - (Optional) The layout that the exported timestamp will use. It must follow the [Golang format](https://golang.org/pkg/time/#pkg-constants). If this parameter is provided, `timezone_layout` will be ignored. Otherwise the exported timestamp will use the same layout as in `local_time`, plus `timezone_layout`.

Reviewer: ```suggestion
- `layout` - (Optional) The layout that the exported timestamp will use. It must follow the [Golang format](https://golang.org/pkg/time/#pkg-constants). This attribute conflicts with `timezone_layout`.
```

Author: @bendrucker Do you want the behavior to be a conflict rather than ignore?

Reviewer: Yes, attributes that are populated should never be ignored. They should either be consumed or rejected with a validation error.

Reviewer: Also, avoid describing behavior when the field isn't set if there's a conflicting field. Put that description in the conflicting field.

> - `timezone_layout` - (Optional) The timezone part of `layout`. It's useless when `layout` is provided. Default to " -0700".

Reviewer: Why is this a separate attribute at all? Why can't the time zone be inserted into the layout?

Reviewer: Also, -0700 is an arbitrary default. If this field needs to exist at all, it should be user supplied with no default

Author: @bendrucker The idea is to provide an easy way to keep the existing layout of local time and add a timezone. You'll have a better idea after seeing the example in the doc. It has to have a default since local_time does not provide a layout for the timezone part

Reviewer: Ok, I think I get it. The implementation is leaking.

A few things come to mind:

* Trim the leading space
* Call it `layout_timezone`
* Clarify that this is a layout, and not a literal time zone value. Give examples of equivalent layouts to illustrate this point, e.g., `-0700` and `MST`.
* Use formal schema language (e.g. "conflicts") not informal statements like "it's useless." And explain how it conflicts with the other field (layout), i.e. that this value should be included as part of `layout` if specified
